src/prediction_market/views/pred_market_votes_view.py

```diff
@@ -0,0 +1,181 @@
+                prediction_market=prediction_market,
+                vote=vote,
+                bet_amount=bet_amount,
+            )
+
+            # track as contribution if it's a new vote
+            create_contribution.apply_async(
+                (
+                    Contribution.REPLICATION_VOTE,
+                    {
+                        "app_label": "prediction_market",
+                        "model": "predictionmarketvote",
+                    },
+                    request.user.id,
+                    prediction_market.unified_document.id,
+                    prediction_market_vote.id,
+                ),
+                priority=1,
+                countdown=10,
+            )
+
+        context = self._get_retrieve_context()
+        data = DynamicPredictionMarketVoteSerializer(
+            prediction_market_vote, context=context
+        ).data
+        return Response(data, status=200)
+
+    def _get_retrieve_context(self):
+        context = self.get_serializer_context()
+        context = {
+            **context,
+            "rhc_dcs_get_created_by": {
+                "_include_fields": (
+                    "id",
+                    "author_profile",
+                    "first_name",
+                    "last_name",
+                    "editor_of",
+                )
+            },
+            "usr_dus_get_author_profile": {
+                "_include_fields": (
+                    "id",
+                    "first_name",
+                    "last_name",
+                    "created_date",
+                    "updated_date",
+                    "profile_image",
+                )
+            },
+        }
+        return context
+
+    def list(self, request, *args, **kwargs):
+        pred_mkt_id = request.query_params.get("prediction_market_id")
+        is_user_vote = request.query_params.get("is_user_vote")
+
+        user = request.user
+
+        if pred_mkt_id is None:
+            return Response({"message": "prediction_market_id is required"}, status=400)
+        try:
+            prediction_market = PredictionMarket.objects.get(id=pred_mkt_id)
+        except PredictionMarket.DoesNotExist:
+            return Response({"message": "Prediction market does not exist"}, status=400)
+        queryset = PredictionMarketVote.objects.filter(
+            prediction_market=prediction_market
+        )
+
+        if is_user_vote is not None:
+            queryset = queryset.filter(created_by=user)
+
+        # Apply ordering
+        ordering = OrderingFilter().get_ordering(request, queryset, self)
+        if ordering:
+            queryset = queryset.order_by(*ordering)
+
+        context = self._get_retrieve_context()
+        serializer = DynamicPredictionMarketVoteSerializer(
+            queryset, many=True, context=context
+        )
+        return Response(serializer.data)
+
+    def destroy(self, request, *args, **kwargs):
+        user = request.user
```

Comment:
Instead of this being a true delete, would it be better to have a status="neutral" or something on the vote? That way it's more of a "soft delete" and also keeps one item in the DB for a "Yes", "No", or "Remove Vote"

Reply:
Good point! Made it a soft delete.